app/models/workers_scaling/experiment_resources_interface.rb

```diff
@@ -157,15 +161,20 @@ def yield_workers(records)
     private
 
     ##
-    # Compares infrastructure description from #get_available_infrastructures with @allowed_infrastructures entry.
-    # Entry from @allowed_infrastructures may have additional fields that
-    # should not be taken into consideration in comparison.
+    # TODO: update description after merge with WSS-35 - enabled inf
+    # Compares whether all fields from narrower are equal with corresponding fields from wider
+    # when exact flag is set to false. Performs full comparison when exact flag is true.
+    # By default exact flag is set to false
     # Return true when infrastructures are equal, false otherwise.
-    def infrastructures_equal?(from_available, from_allowed)
+    def infrastructures_equal?(narrower, wider, exact=false)
       # TODO replace with infrastructure id
-      return false if from_allowed[:name] != from_available[:name]
-      from_available[:params].each do |key, value|
-        return false if from_allowed[:params][key] != value
+      return false if wider[:name] != narrower[:name]
+      if exact
+        return false unless wider[:params] == wider[:params]
```

Comment:
Czy w tym miejscu wydarzy się efektywnie cokolwiek innego, niż w else? Mam na myśli, czy porównanie hashy przez == jest wydajniejsze. Jeżeli nie, to nie bardzo widzę sens tego rozgraniczenia.